sdk/helper/testhelpers/schema/response_validation.go

```diff
@@ -126,3 +127,43 @@ func GetResponseSchema(t *testing.T, path *framework.Path, operation logical.Ope
 
 	return &schemaResponses[0]
 }
+
+// ResponseValidatingCallback can be used in setting up a [vault.TestCluster] that validates every response against the
+// openapi specifications
+//
+// [vault.TestCluster]: https://pkg.go.dev/github.com/hashicorp/vault/vault#TestCluster
+func ResponseValidatingCallback(t *testing.T) func(logical.Backend, *logical.Request, *logical.Response) {
+	type PathRouter interface {
```

Comment:
interesting workaround here.

Reply:
_sigh_ yeah... couldn't think of a better way to do it 🤷

Reply:
I like this implementation. It reminds of something similar I saw in the implementation of Go's context. 👍